packages/ice/src/turn/protocol.ts

```diff
@@ -201,14 +213,20 @@ class TurnClient implements Protocol {
   refresh = () =>
     new PCancelable(async (r, f, onCancel) => {
       let run = true;
+      let timeoutHandle: NodeJS.Timeout | undefined;
       onCancel(() => {
         run = false;
+        if (timeoutHandle !== undefined) {
+          clearTimeout(timeoutHandle);
+        }
         f("cancel");
       });
 
       while (run) {
         // refresh before expire
-        await setTimeout((5 / 6) * this.lifetime * 1000);
+        await new Promise<void>((resolve) => {
+          timeoutHandle = setTimeout(resolve, (5 / 6) * this.lifetime * 1000);
```

Comment:
So, if we clear the timeout, does resolve get called?

Reply:
Replace with an AbortController !